assets/src/components/ladder.tsx

```diff
@@ -152,7 +159,7 @@ const VehicleSvg = ({
       <g
         className={`m-ladder__vehicle ${selectedClass}`}
         transform={`translate(${x},${y})`}
-        onClick={() => dispatch(selectVehicle(vehicleId))}
+        onClick={() => dispatch(selectVehicle(associatedVehicleId(vehicleId)))}
```

Comment:
I would be very opposed to that. The entire point of an ID is to be unique, giving multiple entities the same ID would violate that basic principle.

Reply:
The thing that's being uniquely identified is the vehicle, not the icon. We already show multiple icons for the same vehicle when vehicles are incoming from another route (it's a ghost in this screenshot but we do the same for corporeal vehicles). The fact that they share ids because they're referencing the same vehicle means that they get highlighted at the same time, which is what we want in this case, too.
<img width="883" alt="Screen Shot 2020-01-23 at 16 42 40" src="https://user-images.githubusercontent.com/23065557/73026546-9856c680-3dff-11ea-990a-9c65069ed3b4.png">

Reply:
I wasn't referring to icons. We have a collection of "vehicle" data and my understanding is that you are proposing giving multiple items in that collection of data the same ID. That is what I'm saying is a bad idea.

Reply:
Yeah, "icon" was probably not the right word for what I meant. I guess "object" might be better. The id is identifying the physical vehicle, not identifying the object. So it's okay if two objects have the same id, as long as two different buses don't.

Here, the two objects both refer to the same vehicle. Just like how for an incoming bus the object for the incoming box of the new route refers to the same vehicle as the object that's on_route on the previous route.

We probably even want two objects referring to the same vehicle to have the same id, so highlighting works well.